project/SettingsPublish.scala

```diff
@@ -38,8 +40,8 @@ trait SettingsPublish {
     },
     credentials += Credentials("Sonatype Nexus Repository Manager",
       "oss.sonatype.org",
-      sys.env("CI_DEPLOY_USERNAME"),
-      sys.env("CI_DEPLOY_PASSWORD")),
+      sys.env.getOrElse("CI_DEPLOY_USERNAME", ""),
+      sys.env.getOrElse("CI_DEPLOY_PASSWORD", "")),
```

Comment:
I think that this is better because SBT compiles this code on load. If another user clones the repo they need to provide these variables even though they don't want to execute theses tasks.